server/src/main/java/org/elasticsearch/action/admin/indices/sampling/GetSampleConfigurationAction.java

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.action.admin.indices.sampling;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.ActionResponse;
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.action.IndicesRequest;
+import org.elasticsearch.action.support.IndicesOptions;
+import org.elasticsearch.action.support.local.LocalClusterStateRequest;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.xcontent.ChunkedToXContentObject;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * Action type for retrieving sampling configuration for a specific index.
+ * <p>
+ * This action allows clients to get the current sampling configuration
+ * that has been set on a specific index. This action
+ * targets a single index and returns its configuration in a structured format.
+ * </p>
+ * The response format matches:
+ *  <pre>
+ * {
+ *  "index": "logs",
+ *  "configuration": {
+ *      "rate": ".5",
+ *      "if": "ctx?.network?.name == 'Guest'"
+ *  }
+ * }
+ *  </pre>
+ *
+ * @see SamplingConfiguration
+ * @see PutSampleConfigurationAction
+ */
+public class GetSampleConfigurationAction extends ActionType<GetSampleConfigurationAction.Response> {
+
+    /**
+     * Singleton instance of the GetSampleConfigurationAction.
+     * This provides a shared reference to the action type throughout the application.
+     */
+    public static final GetSampleConfigurationAction INSTANCE = new GetSampleConfigurationAction();
+
+    /**
+     * The name identifier for this action type used in the transport layer.
+     */
+    public static final String NAME = "indices:admin/sample/config/get";
```

Comment:
I think we want this to be `"indices:monitor/sample/config/get"`

Reply:
The permissions are based off of it. We want fetching the config to only require the index monitor permission (vs index admin to create or modify or delete it).